### DISCUSSION TITLE
Switch CNI for existing cluster

### DISCUSSION THREAD

**Discussion by fischerman** [rancher/rke2#2320] [Q&A] [answered]
Can I switch the CNI on an existing cluster by simply setting the `cni` config field? What happens with existing nodes / pods?

**Comment by brandond** [comment, marked as answer]
No, that is not supported. The CNI should be selected once, when the cluster is initially being created. If you feel like doing it on your own you may be able to hack at it and get it to work, but I would not expect it to be particularly easy.
